### PR TITLE
change Http to HttpClient

### PR DESCRIPTION
**Pull request by agustinvinao**
HttpClient implementation. Part of the changes needs to update from Headers to HttpHeaders.

### CODE REVIEW

**Comment by jamesdixon** [comment]
Any chance this gets merged soon?

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
   },
   "repository": {
     "type": "git",
-    "url": "https://github.com/ghidoz/angular2-jsonapi"
+    "url": "https://github.com/agustinvinao/angular2-jsonapi"
```
Comment (HennerM) [inline package.json]: can you revert this change please?

File: package.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
   ],
   "license": "MIT",
   "bugs": {
-    "url": "https://github.com/ghidoz/angular2-jsonapi/issues"
+    "url": "https://github.com/agustinvinao/angular2-jsonapi/issues"
```
Comment (HennerM) [inline package.json]: this url as well

File: src/decorators/has-one.decorator.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export function HasOne(config: any = {}) {
```
Comment (HennerM) [inline src/decorators/has-one.decorator.ts]: What distinguishes HasOne From `BelongsTo´ ?

File: test/fixtures/comment.fixture.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+// import {BOOK_PUBLISHED, BOOK_TITLE} from "./author.fixture";
```
Comment (HennerM) [inline test/fixtures/comment.fixture.ts]: Is this file still needed?

File: src/services/json-api-datastore.service.ts [hunk 5]
```diff
@@ -161,18 +162,18 @@ export class JsonApiDatastore {
     }
 
     private extractQueryData<T extends JsonApiModel>(res: any, modelType: ModelType<T>, withMeta = false): T[] | JsonApiQueryData<T> {
-        let body: any = res.json();
+        let body: any = res;
         let models: T[] = [];
-        body.data.forEach((data: any) => {
-            let model: T = new modelType(this, data);
+        let model: T;
+        body.data.map((_data: any) => {
```
Comment (HennerM) [inline src/services/json-api-datastore.service.ts]: can this bet written as
´models = body.data.map( ... )` 
and `return model` in the map function?

File: src/services/json-api-datastore.service.ts [hunk 6]
```diff
@@ -225,22 +226,17 @@ export class JsonApiDatastore {
         return jsonApiMeta;
     }
 
-    private getOptions(customHeaders?: Headers): RequestOptions {
-        let requestHeaders = new Headers();
-        requestHeaders.set('Accept', 'application/vnd.api+json');
-        requestHeaders.set('Content-Type', 'application/vnd.api+json');
-        if (this._headers) {
```
Comment (HennerM) [inline src/services/json-api-datastore.service.ts]: Removing this means the user can not set "global" headers anymore. Was this intended?

**Comment by agustinvinao** [comment]
Im working to check your comments. I'll update the PL as soon as I can. I've forked the repo and Im using it with this changes, i need to see if all your comments still work on my current project, mainly see if the globals headers works ok.

**Comment by safo6m** [comment]
@agustinvinao you are changing a lot of different things in this pull request. It will be a lot easier and safer for merging if you focus on one change at a time. Can you please create a few smaller pull requests? 
Switching to HttpClient would be a great start.
